File: Utils/FileUtils.m
#import "FileUtils.h"

@implementation FileUtils

+(NSString *) getDocumentPath {
    NSArray * paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString * document = [paths objectAtIndex:0];
    return document;
}

+(NSString *) makeDir: (NSString *) dir basePath: (NSString *) path {
    NSString * pathTmp = [path stringByAppendingPathComponent:dir];
    NSFileManager * fmgr = [NSFileManager defaultManager];
    if (![fmgr fileExistsAtPath:pathTmp]) {
        [fmgr createDirectoryAtPath:pathTmp withIntermediateDirectories:YES attributes:nil error:nil];
    }
    return pathTmp;
}

+(void) writeTextFile:(NSString *)fileName savePath:(NSString *)path fileContent:(NSString *)text {
    NSString * document = [self getDocumentPath];
    NSString * pathTmp = [self makeDir:path basePath:document];
    NSString * fileOper = [pathTmp stringByAppendingPathComponent:fileName];
    NSData * writeData = [text dataUsingEncoding:NSUTF8StringEncoding];
    [writeData writeToFile:fileOper atomically:YES];
}

+(NSString *) readTextFile:(NSString *)fileName loadPath:(NSString *)path {
    NSString * document = [self getDocumentPath];
    NSString * pathTemp = [document stringByAppendingPathComponent:path];
    NSString * fileOper = [pathTemp stringByAppendingPathComponent:fileName];
    
    NSString * text = @"";
    if ([self fileExists:fileName filePath:path]) {
        NSData * readData = [NSData dataWithContentsOfFile:fileOper];
        text = [[NSString alloc] initWithData:readData encoding:NSUTF8StringEncoding];
    }
    
    return text;
}

+(void) writeFile:(NSString *)fileName savePath:(NSString *)path fileData:(NSData *)data {
    NSString * document = [self getDocumentPath];
    NSString * pathTmp = [self makeDir:path basePath:document];
    NSString * fileOper = [pathTmp stringByAppendingPathComponent:fileName];
    [data writeToFile:fileOper atomically:YES];
}

+(NSData *) readFile:(NSString *)fileName loadPath:(NSString *)path {
    NSString * document = [self getDocumentPath];
    NSString * pathTemp = [document stringByAppendingPathComponent:path];
    NSString * fileOper = [pathTemp stringByAppendingPathComponent:fileName];
    NSData * retData = nil;
    if ([self fileExists:fileName filePath:path]) {
        retData = [NSData dataWithContentsOfFile:fileOper];
    }
    return retData;
}

+(BOOL) fileExists:(NSString *)fileName filePath:(NSString *)path {
    NSString * document = [self getDocumentPath];
    NSString * pathTemp = [document stringByAppendingPathComponent:path];
    NSString * fileOper = [pathTemp stringByAppendingPathComponent:fileName];
    NSFileManager * fmgr = [NSFileManager defaultManager];
    return [fmgr fileExistsAtPath:fileOper];
}

+ (long long) fileSizeAtPath:(NSString*) filePath{
    NSFileManager* manager = [NSFileManager defaultManager];
    if ([manager fileExistsAtPath:filePath]){
        return [[manager attributesOfItemAtPath:filePath error:nil] fileSize];
    }
    return 0;
}

+ (float) folderSizeAtPath:(NSString*) folderPath{
    NSFileManager* manager = [NSFileManager defaultManager];
    if (![manager fileExistsAtPath:folderPath]) return 0;
    NSEnumerator *childFilesEnumerator = [[manager subpathsAtPath:folderPath] objectEnumerator];
    NSString* fileName;
    long long folderSize = 0;
    while ((fileName = [childFilesEnumerator nextObject]) != nil){
        NSString* fileAbsolutePath = [folderPath stringByAppendingPathComponent:fileName];
        folderSize += [self fileSizeAtPath:fileAbsolutePath];
    }
    return folderSize/(1024.0*1024.0);
}

@end

File: swift/Utils/FileUtils.swift
import UIKit

class FileUtils: NSObject {
    
    class func getDocumentPath() -> String {
        var paths = NSSearchPathForDirectoriesInDomains(NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomainMask.AllDomainsMask, true);
        return paths[0] 
    }
    
    class func makeDir(dir: String, basePath path: String) -> String {
        let pathTmp = "\(path)/\(dir)"
        let fmgr = NSFileManager.defaultManager()
        
        if (!fmgr.fileExistsAtPath(pathTmp)) {
            do {
                try fmgr.createDirectoryAtPath(pathTmp, withIntermediateDirectories:true, attributes:nil)
            } catch _ {
            }
        }
        return pathTmp
    }
    
    class func writeTextFile(fileName: String, savePath path: String, fileContent text: String) {
        let document = self.getDocumentPath()
        let pathTmp = self.makeDir(path, basePath:document)
        let fileOper = "\(pathTmp)/\(fileName)"
        let writeData = (text as NSString).dataUsingEncoding(NSUTF8StringEncoding)
        writeData!.writeToFile(fileOper, atomically:true)
    }
    
    class func readTextFile(fileName: String, loadPath path: String) -> String {
        let document = self.getDocumentPath()
        let pathTemp = "\(document)/\(path)"
        let fileOper = "\(pathTemp)/\(fileName)"
    
        var text: String = ""
        if (self.fileExists(fileName, filePath:path)) {
            let readData = NSData(contentsOfFile: fileOper)
            text = NSString(data: readData!, encoding: NSUTF8StringEncoding) as! String
        }
        return text
    }
    
    class func writeFile(fileName: String, savePath path: String, fileData data: NSData) {
        let document = self.getDocumentPath()
        let pathTmp = self.makeDir(path, basePath:document)
        let fileOper = "\(pathTmp)/\(fileName)"
        data.writeToFile(fileOper, atomically:true)
    }
    
    class func readFile(fileName: String, loadPath path: String) -> NSData? {
        let document = self.getDocumentPath()
        let pathTemp = "\(document)/\(path)"
        let fileOper = "\(pathTemp)/\(fileName)"
        var retData: NSData? = nil
        if (self.fileExists(fileName, filePath:path)) {
            retData = NSData(contentsOfFile: fileOper)
        }
        return retData
    }
    
    class func fileExists(fileName: String, filePath path: String) -> Bool {
        let document = self.getDocumentPath()
        let pathTemp = "\(document)/\(path)"
        let fileOper = "\(pathTemp)/\(fileName)"
        let fmgr = NSFileManager.defaultManager()
        return fmgr.fileExistsAtPath(fileOper)
    }
    
    class func fileSizeAtPath(filePath: String) -> UInt64 {
        let manager = NSFileManager.defaultManager()
        if (manager.fileExistsAtPath(filePath)) {
            let attrs = try? manager.attributesOfItemAtPath(filePath)
            let dic = NSDictionary(dictionary: attrs!)
            return dic.fileSize()
        }
        return 0
    }
    
    class func folderSizeAtPath(folderPath: String) -> Double {
        let manager = NSFileManager.defaultManager()
        if (!manager.fileExistsAtPath(folderPath)) {
            return 0
        }
        let childFilesEnumerator = manager.subpathsAtPath(folderPath)
        var folderSize: UInt64 = 0
        if (childFilesEnumerator != nil) {
            for fileName in childFilesEnumerator! {
                let fileAbsolutePath = "\(folderPath)/\(fileName)"
                folderSize += self.fileSizeAtPath(fileAbsolutePath)
            }
        }
        // Byte to M
        let fz: Double = Double(folderSize)
        return (fz / 1024.0 / 1024.0)
    }
   
}
